src/transformers/pipelines/mask_generation.py

```diff
@@ -226,7 +238,13 @@ def _forward(
         original_sizes = model_inputs.pop("original_sizes").tolist()
         reshaped_input_sizes = model_inputs.pop("reshaped_input_sizes").tolist()
 
-        model_outputs = self.model(**model_inputs)
+        intermediate_embeddings = model_inputs.pop("intermediate_embeddings", None)
+        forward_params = inspect.signature(self.model.forward).parameters
+
+        if "intermediate_embeddings" in forward_params and intermediate_embeddings is not None:
+            model_outputs = self.model(**model_inputs, intermediate_embeddings=intermediate_embeddings)
+        else:
+            model_outputs = self.model(**model_inputs)
```

Comment:
as per comment this should work ootb
```suggestion
        model_outputs = self.model(**model_inputs)
```

Reply:
added the suggestion no tests are breaking